src/ComposerRequireChecker/Cli/ResultsWriter/BatchWriter.php

```diff
@@ -0,0 +1,31 @@
+<?php
+
+declare(strict_types=1);
+
+namespace ComposerRequireChecker\Cli\ResultsWriter;
+
+use function array_filter;
+
+final class BatchWriter implements ResultsWriter
+{
+    /** @var ResultsWriter[] */
+    private array $writer;
+
+    public function __construct(ResultsWriter ...$writer)
+    {
+        $this->writer = array_filter(
+            $writer,
+            static fn (ResultsWriter $writer) => ! ($writer instanceof BatchWriter)
```

Comment:
This callback is not necessary - it is impossible to pass in instances that aren't `BatchWriter` (this codebase really needs `infection/infection` @_@ )